src/compas_cgal/straight_skeleton_2.py

```diff
@@ -155,6 +155,63 @@ def create_offset_polygons_2(points, offset) -> list[Polygon]:
     return [Polygon(points.tolist()) for points in offset_polygons]
 
 
+def create_offset_polygons_with_holes_2(points, holes, offset) -> list[Tuple[Polygon, list[Polygon]]]:
+    """Compute the polygon offset with holes.
+
+    Parameters
+    ----------
+    points : list of point coordinates or :class:`compas.geometry.Polygon`
+        The points of the polygon.
+    holes : list of list of point coordinates or list of :class:`compas.geometry.Polygon`
+        The holes of the polygon.
+    offset : float
+        The offset distance. If negative, the offset is outside the polygon, otherwise inside.
+
+    Returns
+    -------
+
+    Returns
+    -------
+    list of tuple of (:class:`Polygon`, list[:class:`Polygon`])
+        The polygons with holes.
+
+    Raises
+    ------
+    ValueError
+        If the normal of the polygon is not [0, 0, 1].
+        If the normal of a hole is not [0, 0, -1].
+    """
+    points = list(points)
+    normal = normal_polygon(points, True)
+    if not TOL.is_allclose(normal, [0, 0, 1]):
+        raise ValueError("The normal of the polygon should be [0, 0, 1]. The normal of the provided polygon is {}".format(normal))
+    V = np.asarray(points, dtype=np.float64)
+
+    H = []
+    for i, hole in enumerate(holes):
+        points = list(hole)
+        normal_hole = normal_polygon(points, True)
+        if not TOL.is_allclose(normal_hole, [0, 0, -1]):
+            raise ValueError("The normal of the hole should be [0, 0, -1]. The normal of the provided {}-th hole is {}".format(i, normal_hole))
+        hole = np.asarray(points, dtype=np.float64)
+        H.append(hole)
+
+    offset = float(offset)
```

Comment:
why the conversion?

Reply:
in any case, would do this at the beginning of the algorithm. process all input params (if necessary) at the start...

Reply:
you are right, it is not needed